frontend/components/tree/js/tree.service.ts

```diff
@@ -1131,10 +1138,31 @@ export class TreeService {
 
 		}
 
+		const lastNode = nodes[nodes.length - 1] ;
+		this.handleMetadata(lastNode);
+
 		return this.highlightNodes(nodes, multi, colour);
 
 	}
 
+	/**
+	 * Show metadata in the metadata panel if necessary
+	 * @param lastNode the node to show the metadata for
+	 */
+	public handleMetadata(lastNode) {
+		// Handle showing metadata in the metadata panel if necessary
+
+		if (lastNode && lastNode.meta) {
+			if (this.DocsService.state.active && !this.ViewerService.pin.pinDropMode) {
+				this.DocsService.handleObjectSelected(
```

Comment:
This is potentailly badly named now that what we are doing has changed.

Should be called displayDocument or something